### ISSUE TITLE
Event Listener on value when using orderByChild()

### ISSUE THREAD

**Issue by bakr-baroudi** [invertase/react-native-firebase#43]
Hi guys,
Thanks for such package. I appreciate your effort on it.

I am trying to fetch a certain data from my firebase realtime database. I am running very simple query which is under orderByChild().

```javascript 
   queryByMonth(startDate,endDate) {
		this.userRef.orderByChild('timestamp').startAt(startDate).endAt(endDate)
		.on("value", (snap) => {
			console.warn(snap.val())
			if (snap.val()) {
				this.setState({
				dataSource: this.state.dataSource.cloneWithRows(snap.val())
			});
		} else {
				this.setState({
				dataSource: this.state.dataSource.cloneWithRows([])
			});
		}
	});
     }
```
* The issue here is, for Ex: 

- First query: 
Starting day is '1490976000000' which is : 01 April 2017,
Ending day is '1493567999000' which is : 30 April 2017,
Expected result : (only records that is between  01 April & 30 April). 2 Records
This event get triggered and I am getting the correct result.

- Second query:
Starting day is '1490976000000' which is : 01 April 2017,
Ending day is '1496159999000' which is : 30 May 2017,
Expected result : (only records that is between  01 April & 30 May). 3 Records
This event get triggered and I am getting the correct result. 

- Third query:
Starting day is '1490976000000' which is : 01 April 2017,
Ending day is '1493567999000' which is : 30 April 2017,
Expected result : (only records that is between  01 April & 30 April). 2 Records
This event WILL NOT trigger and I do not get snapshot. Thats means data still 3 records but I expected 2 records as the first query result.

NOTE: If I selected less or equal to 29 April 2017 for the ending date of the third query, the event get triggered and i will get the correct result which is 2 records!!

Any Idea?

**Comment by chrisbianca** [comment]
I think this is the same bug as https://github.com/invertase/react-native-firebase/issues/25

The third query is the same as the first query. Unfortunately, due to the way RNFirebase is written to attempt to optimise the data being sent between native and JS it means that whenever a ref or query that has already been created is accessed a second time, it does not re-send the initial data.

A fix for this is on the work plan, but still a little way off as it requires a reasonable rewrite of the underlying database modules for both iOS and Android.

cc/ @Salakar @Ehesp to highlight another place where this issue has popped up.

**Comment by bakr-baroudi** [comment]
Thanks @chrisbianca  for your reply.
Any temporary solution currently?

**Comment by AndrewHenderson** [comment]
@bakr-baroudi You could use something like Redux for caching client-side snapshots.

This would mean some extra code for client-side data management which is why I've held off on doing this myself.

Hoping this fix gets released soon.

**Comment by bakr-baroudi** [comment]
Hi @AndrewHenderson, I am using Mobx for caching snapshots data. But I am running in a different issue related to successful callback which is the code keep running and does not wait for snapshots data to filled in mobx store.

**Comment by chrisbianca** [comment]
This should be resolved as of 1.0.0-alpha13.
Please let us know if you're still having issues once you've updated.
